jre_emul/Classes/com/google/j2objc/net/IosHttpsURLConnection.java

```diff
@@ -60,15 +60,22 @@ public String getCipherSuite() {
     return new Certificate[0];
   }
 
+  // Delegate methods.
   @Override
   public Certificate[] getServerCertificates() throws SSLPeerUnverifiedException {
-    // TODO(tball): implement
-    logger.severe("HttpsURLConnection.getServerCertificates() not implemented");
-    return new Certificate[0];
+    if (getURL().getHost().startsWith("http://")) {
+      throw new SSLPeerUnverifiedException("The http protocol does not support certificates");
+    }
+      
+    List<Certificate> certificates = delegate.certificates;
+    
+    if (certificates != null && !certificates.isEmpty()) {
+      return certificates.toArray(new Certificate[certificates.size()]);
+    }
+    
+    return null;
```

Comment:
The normal JRE pattern is to return an empty array (new Certificate[0]) instead of null for any method that returns an array, unless documented otherwise.